src/printer.js

```diff
@@ -3260,6 +3273,84 @@ function hasLeadingOwnLineComment(text, node) {
   return res;
 }
 
+function hasNakedLeftSide(node) {
+  return (
+    node.type === "BinaryExpression" ||
```

Comment:
I can't remember the context but I probably meant that it was ugly to enumerate all those node types, rather than the way we're actually doing it :)